### DISCUSSION TITLE
safe.directory issues

### DISCUSSION THREAD

**Discussion by derff44** [git-for-windows/git#3987] [Q&A]
Deployment of git version 2.37.2.windows.1 to our QA environment led to an interesting issue.  I am not sure if this is a bug or not, so I thought Id ask here first. 

Running from CMD on a windows server 2016
Pointing all users config file to C:\Program Files\Git\etc\gitconfig

As expected we started to receive unsafe directory errors from our Jenkins jobs.  We started to add paths directly into the C:\Program Files\Git\etc\gitconfig file, and tested after each addition.  Jobs started running correctly.  Unfortunately when we started to get down to over 100 lines of paths under the [safe] block, new jobs would no longer recognize the directories as safe.  Testing the first jobs we started with still returned positive results. 

Another thing we noticed, when running `git config --list --show-origin`, the output got to about line 40 and then halted.  Hitting enter key would sometimes load another line or two, but eventually we would need to "control C "out of the command. 

My questions are:
- Is there a best practice for adding many safe.directory locations in windows?
- Is there a limit to the size and/or line items git can read from the C:/Program Files/Git/mingw64/etc/gitconfig file?
- Are there any plans to allow wildcards so we can just add top directories and have the safe.directory setting apply recursively?

Thank you!

**Comment by rimrul** [comment]
> Is there a best practice for adding many safe.directory locations in windows?

The only best practice there is is "try to avoid having to do that",  but that's probably not what you're looking for.

> Is there a limit to the size and/or line items git can read from the C:/Program Files/Git/mingw64/etc/gitconfig file?

Yes. From what I can tell it's `INT_MAX - 1` (2147483646) lines accross `system` and `global` config files. At least for reading this so called protected config.

> Are there any plans to allow wildcards so we can just add top directories and have the safe.directory setting apply recursively?

No current plans for wildcards, no. There is `safe.directory=*` as a special value, but that's about it.

**Reply by derff44** [reply]
I was afraid you were going to say that.  This feature seems.... Less than thought through.  But, I appreciate you taking the time to reply.

**Comment by mfriedrich74** [comment]
Hi,

Is there a reason preventing to switch from group ownership to a single
Jenkins service account as owner for the repo / work root folder? I would
think a single user as owner would solve issue, if that's a feasible
solution.

Best regards, Mike
